docs/source/contributing/developing.adoc

```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
 However, for cases which require further investigation, the integration test also logs more detailed progress into a log file.
 This file is located at *_$GOPATH/github.com/minishift/minishift/out/test-results/integration_YYYY-MM-DD_HH:MM:SS.log_*.
 
+[[writing-features]]
+==== Writing features
+
+Minishift integration testsuite offers comprehensive set of implemented Gherkin steps which allow for comfortable writing of feature files.
+All available steps are defined in file link:https://github.com/minishift/minishift/blob/master/test/integration/testsuite/testsuite.go[/test/integration/testsuite.go].
+Definitions of steps are simple and self-documenting, for example:
+
+----
+s.Step(`^executing "minishift (.*)" (succeeds|fails)$`,
+    ExecutingMinishiftCommandSucceedsOrFails)
+----
+
+defines a step which matches regular expression `^executing "minishift (.*)" (succeeds|fails)$`. If matched the two capturing groups are passed as parameters to the function `ExecutingMinishiftCommandSucceedsOrFails(command string, expectedResult string)`, which implements the actual behaviour of the step.
+
+To use the step in the feature file, you need to make sure it matches with the regular expression and prepend one of the Gherkin keywords: `Given`, `When`, `Then`, `And` or `But`, for example:
+
+----
+Then executing "minishift config set" succeeds
```

Comment:
Example is not correct. It won't work in normal condition due to the missing ```PROPERTY_NAME``` and ```PROPERTY_VALUE```. We have a separate implementation for setting config in the testsuite.go. I think use the same steps as above so that it will be in sync 
```
When executing "minishift start" succeeds
```

Reply:
@amitkrout Good catch, will fix it, thanks.